jvm-packages/xgboost4j-spark-gpu/src/main/java/ml/dmlc/xgboost4j/java/spark/GpuColumnVector.java

```diff
@@ -0,0 +1,589 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.java.spark;
+
+import java.util.Arrays;
+import java.util.List;
+
+import ai.rapids.cudf.ColumnView;
+import ai.rapids.cudf.DType;
+import ai.rapids.cudf.HostColumnVector;
+import ai.rapids.cudf.Scalar;
+import ai.rapids.cudf.Schema;
+import ai.rapids.cudf.Table;
+
+import org.apache.spark.sql.catalyst.expressions.Attribute;
+import org.apache.spark.sql.types.*;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+
+
+/**
+ * A GPU accelerated version of the Spark ColumnVector.
+ * Most of the standard Spark APIs should never be called, as they assume that the data
+ * is on the host, and we want to keep as much of the data on the device as possible.
+ * We also provide GPU accelerated versions of the transitions to and from rows.
+ */
+public class GpuColumnVector extends GpuColumnVectorBase {
```

Comment:
Why are these things on cuDF instead?  Seems weird to have so many data processing code in a machine learning library.

Reply:
This is trivial, I will remove this since we have defined this in the plugin.